File: HC3/Clase4/fibo.m
function [f] = fibo(n)
% FIBO Retorna la sucesion de fibonacci

if n <= 0 
    f = 0;
else
    f = zeros(1, n);
    if n >= 1
        f(1) = 1;
    end
    if n >= 2
        f(2) = 1;
    end
    if n>=3
        for k=3:1:n
            f(k) = f(k-1) + f(k-2);
        end
    end
end

end

File: EDD/C11/fibo.c
/**
 * @file fibo.c
 * @author Jose Luis
 * @brief Fibonacci
 * @version 0.1
 * @date 2023-05-23
 * 
 * @copyright Copyright (c) 2023
 * 
 */
#include<stdio.h>

int fibonacci(int n) {
    if(n == 1 || n == 2) return 1;
    return fibonacci(n-1)+fibonacci(n-2);
}

int main()
{
    printf("%i\n",fibonacci(6));
    return 0;
}
